### DISCUSSION TITLE
How are packages imported affected when deleting cell?

### DISCUSSION THREAD

**Discussion by a9lara** [fonsp/Pluto.jl#768] [Questions about using Pluto] [answered]
If I have defined x = 2 in a cell, and then I erase it, then x is no longer define. But, what happens with the imported packages? For example; if 
cell 1: import PyPlot
and then I deleat cell 1 and then:
cell 2: import Plots 
What does Pluto do with all loaded commands of PyPlot?

Regards.

Armando

**Comment by fonsp** [comment, marked as answer]
Good question! Technically, they are still there (it's not possible to fully _unload_ something in Julia), but they are no longer available in your scope.

In the future, we might prompt you to restart the notebook after unloading a package.

For example, an evil package might do this in its loading process:
```julia
Base.sqrt(x::Int64) = "potato"

hello(x) = "soup"
```

After loading the package (using `import Pakcage` in your notebook), calling `sqrt(5)` will return `potato`. After removing the cell containing `import Package`, you can no longer do `Package.hello(123)`, but `sqrt(123)` will still return `potato`. The only (simple) solution is to restart the notebook.

**Reply by a9lara** [reply]
Thank you very much

**Comment by fonsp** [comment]
#844 will detect this situation, and recommend to restart the notebook.

**Reply by a9lara** [reply]
Thank you very much! :)
